clang/lib/Parse/ParseOpenMP.cpp

```diff
@@ -4539,6 +4539,8 @@ bool Parser::ParseOpenMPVarList(OpenMPDirectiveKind DKind,
   bool NeedRParenForLinear = false;
   BalancedDelimiterTracker LinearT(*this, tok::l_paren,
                                    tok::annot_pragma_openmp_end);
+  BalancedDelimiterTracker AllocateT(*this, tok::l_paren,
+                                     tok::annot_pragma_openmp_end);
```

Comment:
Move it exactly where it used (lines 4807-4819)

Reply:
Fixed as suggested.